pkg/api/authz.go

```diff
@@ -235,11 +235,15 @@ func AuthzHandler(ctlr *Controller) mux.MiddlewareFunc {
 func getUsername(r *http.Request) string {
 	// this should work because it was already parsed in authn middleware
 	basicAuth := r.Header.Get("Authorization")
-	s := strings.SplitN(basicAuth, " ", 2) //nolint:gomnd
-	b, _ := base64.StdEncoding.DecodeString(s[1])
-	pair := strings.SplitN(string(b), ":", 2) //nolint:gomnd
+	if basicAuth != "" {
+		s := strings.SplitN(basicAuth, " ", 2) //nolint:gomnd
+		b, _ := base64.StdEncoding.DecodeString(s[1])
```

Comment:
let's treat these errrors now like in authn.go

and also add a test where anonymous is enabled and authz have only default policies (one global ** and another one per repo) and make some requests with and without credentials


another one where anonymous is enabled and authz have users policies and make some requests with and without credentials

try to find corner cases